lib/core/settings/src/types/settings.dart

```diff
@@ -554,6 +567,45 @@ class ImageViewerSettings extends Equatable {
   }
 }
 
+class TallMediaSettings extends Equatable {
+  const TallMediaSettings({
+    required this.enabled,
+  });
+
+  const TallMediaSettings.defaults() : enabled = true;
+
+  TallMediaSettings.fromJson(Map<String, dynamic> json)
+    : enabled = json['enabled'] ?? true;
+
+  final bool enabled;
+
+  TallMediaSettings copyWith({
+    bool? enabled,
+  }) {
+    return TallMediaSettings(
+      enabled: enabled ?? this.enabled,
+    );
+  }
+
+  Map<String, dynamic> toJson() => {
+    'enabled': enabled,
+  };
+
+  @override
+  List<Object> get props => [enabled];
+
+  static const double aspectRatioThreshold = 2.15;
```

Comment:
move these to TallMediaConfigs, pass the config as a option parameter to the above mentioned fromViewport method.